frame/nfts/src/lib.rs

```diff
@@ -0,0 +1,1725 @@
+		/// - `collection`: The identifier of the collection whose item's metadata to set.
+		/// - `maybe_item`: The identifier of the item whose metadata to set.
+		/// - `namespace`: Attribute's namespace.
+		/// - `key`: The key of the attribute.
+		/// - `value`: The value to which to set the attribute.
+		///
+		/// Emits `AttributeSet`.
+		///
+		/// Weight: `O(1)`
+		#[pallet::weight(T::WeightInfo::force_set_attribute())]
+		pub fn force_set_attribute(
+			origin: OriginFor<T>,
+			set_as: Option<T::AccountId>,
+			collection: T::CollectionId,
+			maybe_item: Option<T::ItemId>,
+			namespace: AttributeNamespace<T::AccountId>,
+			key: BoundedVec<u8, T::KeyLimit>,
+			value: BoundedVec<u8, T::ValueLimit>,
+		) -> DispatchResult {
+			T::ForceOrigin::ensure_origin(origin)?;
+			Self::do_force_set_attribute(set_as, collection, maybe_item, namespace, key, value)
+		}
+
+		/// Clear an attribute for a collection or item.
+		///
+		/// Origin must be either `ForceOrigin` or Signed and the sender should be the Owner of the
+		/// `collection`.
+		///
+		/// Any deposit is freed for the collection's owner.
+		///
+		/// - `collection`: The identifier of the collection whose item's metadata to clear.
+		/// - `maybe_item`: The identifier of the item whose metadata to clear.
+		/// - `namespace`: Attribute's namespace.
+		/// - `key`: The key of the attribute.
+		///
+		/// Emits `AttributeCleared`.
+		///
+		/// Weight: `O(1)`
+		#[pallet::weight(T::WeightInfo::clear_attribute())]
+		pub fn clear_attribute(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			maybe_item: Option<T::ItemId>,
+			namespace: AttributeNamespace<T::AccountId>,
+			key: BoundedVec<u8, T::KeyLimit>,
+		) -> DispatchResult {
+			let maybe_check_owner = T::ForceOrigin::try_origin(origin)
+				.map(|_| None)
+				.or_else(|origin| ensure_signed(origin).map(Some).map_err(DispatchError::from))?;
+			Self::do_clear_attribute(maybe_check_owner, collection, maybe_item, namespace, key)
+		}
+
+		/// Approve item's attributes to be changed by a delegated third-party account.
+		///
+		/// Origin must be Signed and must be an owner of the `item`.
+		///
+		/// - `collection`: A collection of the item.
+		/// - `item`: The item that holds attributes.
+		/// - `delegate`: The account to delegate permission to change attributes of the item.
+		///
+		/// Emits `ItemAttributesApprovalAdded` on success.
+		#[pallet::weight(T::WeightInfo::approve_item_attributes())]
+		pub fn approve_item_attributes(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			item: T::ItemId,
+			delegate: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			let delegate = T::Lookup::lookup(delegate)?;
+			Self::do_approve_item_attributes(origin, collection, item, delegate)
+		}
+
+		/// Cancel the previously provided approval to change item's attributes.
+		/// All the previously set attributes by the `delegate` will be removed.
```

Comment:
Really? Wouldn't it be possible that a user wants to keep attributes previously set, but just stop allowing changes?

Reply:
That could be a tricky thing. Let's imagine there is a reputational service that updates your score once per month. You could have a good reputation now, and once you do smth wrong, you could simply disconnect from that service trying to keep the old records.

Reply:
Another moment is the attribute deposits, if we disallow some address to change our attributes, that dapp should still be able to get the deposit back (thus removing the attribute)